src/django/api/views.py

```diff
@@ -3263,3 +3309,110 @@ def retrieve(self, request, pk=None):
     def update(self, request, pk=None):
         self.validate_request(request)
         return super(ApiBlockViewSet, self).update(request, pk=pk)
+
+
+class FacilityActivityReportAutoSchema(AutoSchema):
+    def get_serializer_fields(self, path, method):
+        return [
+            coreapi.Field(
+                name='data',
+                location='body',
+                description=(
+                    'The country, name, and address of the facility. See '
+                    'the sample request body above.'),
+                required=True,
+            )
+        ]
+
+
+def update_facility_activity_report_status(facility_activity_report,
+                                           request, status):
+    status_change_reason = request.data.get('status_change_reason')
+    now = str(datetime.utcnow())
+
+    facility_activity_report.status_change_reason = status_change_reason
+    facility_activity_report.status_change_by = request.user
+    facility_activity_report.status_change_date = now
+    facility_activity_report.status = status
+    if status == 'CONFIRMED':
+        facility_activity_report.approved_at = now
+    facility_activity_report.save()
+
+    return facility_activity_report
+
+
+@schema(FacilityActivityReportAutoSchema())
+class FacilityActivityReportViewSet(viewsets.GenericViewSet):
+    """
+    Approve or reject FacilityActivityReports.
+
+    ## Sample Request Body
+
+        {
+            "status_change_reason": "CLOSED"
+        }
+    """
+    queryset = FacilityActivityReport.objects.all()
+    serializer_class = FacilityActivityReportSerializer
+    permission_classes = (IsRegisteredAndConfirmed,)
+
+    @action(detail=True, methods=['PUT'],
```

Comment:
For consistency with the rest of the API methods we should consider changing the path+verb combination for the approve and reject endpoints.

Pending facility matches have /approve and /reject endpoints that response to POST. It is conventional that "action" endpoints, where the path ends in a word rather than a resource ID, are POST requests.

PUT is actually a reasonable verb choice in this context, because we are updating an existing resource, not adding a new one. If we continue to use PUT, we should consider changing the path to `PUT http://localhost:8081/api/facility-activity-reports/1`.

I don't have strong feelings, but I lean toward changing the verb to POST for consistency with the facility match approve/reject endpoints.

Reply:
I went with `POST` for consistency. Thanks for explaining why it's used in this case!